leaderelection/leaderelection.go

```diff
@@ -158,11 +149,6 @@ func buildConfig(ctx context.Context, log *logp.Logger, client k8s.Interface, bl
 				} else {
 					log.Infof("leader election lock has been acquired by another pod, id: %v", identity)
 				}
-				// called every new beater
-				callOnce.Do(func() {
```

Comment:
`callOnce` can be completely removed from the file.

Reply:
Same goes for `retryOnce`

Reply:
Can you explain why do we need `initOnce`?
What happens after a reconfiguration event from the fleet?

Reply:
As described in the leaderelection file,`retryOnce` is used to re-run the manager in case the leader stopped holding the lease, for example, the pod still exists but the leader has lost the lease due to connectivity issues.

Reply:
`initOnce` is been used to implement a singleton, we want only a single instance of the leader election to exist upon cloudbeat creation.
After a reconfiguration event we call `NewLeaderElector` but the same instance is returned.
A singleton is needed in this case as multiple leader election instances can cause races, especially when we count on the leader election data to trigger the fetching cycle.

Reply:
The `retryOnce` is not required since cloudbeat should try to acquire the lease over and over again and there is no reason to stop after two times.